orderBook.js

```diff
@@ -1,5 +1,32 @@
 function reconcileOrder(existingBook, incomingOrder) {
-  throw new Error('Not Implemented Yet')
+  let updatedBook = []
+
+  if (existingBook.length === 0) {
+    return updatedBook.concat(incomingOrder)
+  }
+
+  if ((existingBook[0].type === 'buy' && incomingOrder.type === 'sell') && (existingBook[0].quantity === incomingOrder.quantity) && (existingBook[0].price === incomingOrder.price)) {
+    return updatedBook
+
+  }
+  else if ((existingBook[0].type === 'buy' && incomingOrder.type === 'sell') && (existingBook[0].quantity > incomingOrder.quantity) && (existingBook[0].price === incomingOrder.price)) {
+    const partialOrder = existingBook[0].quantity - incomingOrder.quantity
+    existingBook[0].quantity = partialOrder
+    return existingBook
+
+  }
+  else if ((existingBook[0].type === 'buy' && incomingOrder.type === 'sell') && (existingBook[0].quantity < incomingOrder.quantity) && (existingBook[0].price === incomingOrder.price)) {
+    const partialSale = incomingOrder.quantity - existingBook[0].quantity
+    incomingOrder.quantity = partialSale
+    updatedBook.push(incomingOrder)
+    return updatedBook
+
+  }
+  else if ((existingBook[0].type === 'buy') && (incomingOrder.type === 'sell') && (existingBook[0].quantity === incomingOrder.quantity) && (incomingOrder.price <= existingBook[0].price)) {
```

Comment:
These ifs only check the first order in the book, but it is possible that there are many orders in the book and the first couple don't match but a later one will.